src/util/locking/RedisLocker.ts

```diff
@@ -17,6 +17,15 @@ const attemptDefaults: Required<AttemptSettings> = { retryCount: -1, retryDelay:
 const PREFIX_RW = '__RW__';
 const PREFIX_LOCK = '__L__';
 
+export interface RedisOptions {
+  /* Username used for AUTH on the Redis server */
+  username?: string;
+  /* Password used for AUTH on the Redis server */
+  password?: string;
+  /* The number of the database to use */
+  db?: number;
+}
+
```

Comment:
I would rename this interface to `RedisSettings`, to be more in line with the `AttemptSettings` argument (in the constructor). It will also lessen any confusion with the `RedisOptions` type of the `ioredis` library.

If you rebase this change onto the main branch, you can also include a `namespacePrefix?: string` property here.